projects/packages/jetpack-mu-wpcom/src/features/domain-email-nag/domain-email-nag.php

```diff
@@ -32,47 +32,69 @@ function get_account_url() {
 }
 
 /**
- * Find the domain, if any, that has an unverified email address.
+ * Get the current domain for the site.
  */
-function get_domain_with_unverified_email() {
-	if ( ! class_exists( 'Domain_Management' ) ) {
-		return false;
-	}
-
-	$domains = \Domain_Management::get_paid_domains_with_icann_verification_status();
+function get_domain() {
+	$urlparts = wp_parse_url( home_url() );
+	return $urlparts['host'];
+}
 
-	foreach ( $domains as $domain ) {
-		if ( $domain['is_pending_icann_verification'] === true ) {
-			return $domain['domain'];
-		}
-	}
-	return false;
+/**
+ * Checks if the site is using a custom domain.
+ */
+function is_custom_domain() {
```

Comment:
This `is_custom_domain` check means we'll only perform the check if the primary domain is custom. I found myself in a situation where my primary address was `*.wordpress.com` but I have a secondary domain which is custom. This check means I don't see the nag even though the API is supposed to be checking all my domains, not just the primary one.

I like the optimisation of trying to only enqueue the nag code when it's necessary, but I don't think we enough information here 🧐

Reply:
I still prefer this as doing an API call every time the user is on the home feels too expensive to me, and I think most people will use their custom domain instead of ours.

Reply:
I guess if they're still using `.wordpress.com` as their primary address it's not as pressing an issue to verify their domain email. It won't break their site to the same extent.